### PR TITLE
odo list component shows components running on podman

### LINKED ISSUE #6297: `odo list` `odo list component` should show components running on podman

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6297 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by feloy** [comment]
Blocked by #6374

File: pkg/podman/version.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	OsArch     string
+	Os         string
+}
+
+type SystemVersionReport struct {
+	Client *Version `json:",omitempty"`
+}
+
+func (o *PodmanCli) Version() (SystemVersionReport, error) {
+	out, err := exec.Command(o.podmanCmd, "version", "--format", "json").Output()
```
Comment (rm3l) [inline pkg/podman/version.go:25]: Wondering if we shouldn't `klog` the actual command that will be launched (here and everywhere we are running `podman`).
I didn't remember I had set `export PODMAN_CMD=docker` on my system some time ago. And the error message (even with increased verbosity) was not really helpful in understanding the error.

```
$ ODO_EXPERIMENTAL_MODE=true odo dev --run-on=podman --debug --no-watch
Experimental mode enabled for flag: --run-on. Use at your own risk. More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
 ✗  unable to access podman. Do you have podman client installed?
```

File: pkg/odo/cli/list/component/list.go [hunk 2]
```diff
@@ -190,7 +217,11 @@ func HumanReadableOutput(list api.ResourcesList) {
 			managedBy = text.Colors{text.FgBlue}.Sprintf(managedBy)
 		}
 
-		t.AppendRow(table.Row{name, componentType, mode, managedBy})
+		row := table.Row{name, componentType, mode, managedBy}
+		if feature.IsEnabled(ctx, feature.GenericRunOnFlag) {
+			row = append(row, comp.RunningOn)
```
Comment (rm3l) [inline pkg/odo/cli/list/component/list.go:222]: To be consistent in the human-readable form, can we display something like `None` when `comp.RunningOn` is empty?

```
$ ODO_EXPERIMENTAL_MODE=true odo list component
Experimental mode enabled for flag: --run-on. Use at your own risk. More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
 ✓  Listing components from namespace 'default' [210ms]
 NAME                                       PROJECT TYPE  RUNNING IN   MANAGED       RUNNING ON 
 my-nodesjs-with-k8s-uri-with-var           nodejs        Dev, Deploy  odo (v3.3.0)  cluster    
 my-nodejs-app-with-composite-run-commands  nodejs        Dev          odo (v3.3.0)  cluster    
 * a-test-nodejs                            Node.js       None         Unknown                  

```

File: pkg/podman/podman.go [hunk 3]
```diff
@@ -71,7 +90,9 @@ func (o *PodmanCli) PlayKube(pod *corev1.Pod) error {
 }
 
 func (o *PodmanCli) PodStop(podname string) error {
-	out, err := exec.Command("podman", "pod", "stop", podname).Output()
+	cmd := exec.Command(o.podmanCmd, "pod", "stop", podname)
+	out, err := cmd.Output()
+	klog.V(3).Infof("executing %v", cmd.Args)
```
Comment (rm3l) [inline pkg/podman/podman.go:95]: Same point about logging before running the command..

File: pkg/podman/podman.go [hunk 4]
```diff
@@ -83,7 +104,9 @@ func (o *PodmanCli) PodStop(podname string) error {
 }
 
 func (o *PodmanCli) PodRm(podname string) error {
-	out, err := exec.Command("podman", "pod", "rm", podname).Output()
+	cmd := exec.Command(o.podmanCmd, "pod", "rm", podname)
+	out, err := cmd.Output()
+	klog.V(3).Infof("executing %v", cmd.Args)
```
Comment (rm3l) [inline pkg/podman/podman.go:109]: Same point about logging before running the command..

File: pkg/podman/podman.go [hunk 5]
```diff
@@ -95,7 +118,9 @@ func (o *PodmanCli) PodRm(podname string) error {
 }
 
 func (o *PodmanCli) VolumeRm(volumeName string) error {
-	out, err := exec.Command("podman", "volume", "rm", volumeName).Output()
+	cmd := exec.Command(o.podmanCmd, "volume", "rm", volumeName)
+	out, err := cmd.Output()
+	klog.V(3).Infof("executing %v", cmd.Args)
```
Comment (rm3l) [inline pkg/podman/podman.go:123]: Same point about logging before running the command..

File: pkg/podman/podman.go [hunk 6]
```diff
@@ -107,7 +132,9 @@ func (o *PodmanCli) VolumeRm(volumeName string) error {
 }
 
 func (o *PodmanCli) VolumeLs() (map[string]bool, error) {
-	out, err := exec.Command("podman", "volume", "ls", "--format", "{{.Name}}", "--noheading").Output()
+	cmd := exec.Command(o.podmanCmd, "volume", "ls", "--format", "{{.Name}}", "--noheading")
+	out, err := cmd.Output()
+	klog.V(3).Infof("executing %v", cmd.Args)
```
Comment (rm3l) [inline pkg/podman/podman.go:137]: Same point about logging before running the command..

**Review by rm3l — CHANGES_REQUESTED** [review]

**Review by rm3l — APPROVED** [review]
Thanks - works great for me!

**Pull request merged**